### ISSUE TITLE
Intuitive way to add a point along the route

### ISSUE THREAD

**Issue by raindropsfromsky** [osmandapp/Osmand#6495]
I have a very common workflow, as follows:

We start with a route, and want to select a POI **along** the route (e.g. pick a place to eat something, or stop to fill up the gas tank, or to draw cash at an ATM). 

These new points are not related to our original route at all. In fact, we hardly give any thought to these points, in terms of planning the trip.

They cannot be the final destination in our route. They would always be waypoints. Furthermore, we will almost never want to reach those points out of turn. (In other words, we simply keep on driving along the route till we come to each spot, make use of that point and move on with our original route.)

So OsmAnd should have a workflow that matches this.
_______________________________________________________________________________________________________
Now I will describe how OsmAnd does not support that workflow.

I set a route, and then selected **Options > Show along the route** option.
Then out of the list of POIs, I chose a particular POI.
OsmAnd shows the chosen point with a marker.

At this point, the OsmAnd should show me only ONE option: **Add to the route**. Obviously I would not want to play with its sequence, such as making it the first waypoint, last waypoint, final destination, etc. 

Instead, Osmand shows me a completely useless display: 
![poi](https://user-images.githubusercontent.com/9047168/52069025-f1509d80-25a3-11e9-968c-4dc5f1f7a3e8.jpg)
It actually seeks to replace the destination with this point. I am not able to do anything with this screen. **[BUG#1]**

Secondly, when I selected the POI in the list, I saw the name of a restaurant (actually that's how I selected it). But somehow Osmand converted that information into LatLong info, which is not usable at all. **[BUG#2]**

Thirdly, there is an **Add+** button on the right, which looks attached to this POI.
So we assume that this button will add the POI to my route.
But when I press this button, the display changes to this:
![addintermediate](https://user-images.githubusercontent.com/9047168/52069078-075e5e00-25a4-11e9-8de0-c04419a3575f.jpg)

In other words, it turns out that this **Add+** button has nothing to do with my chosen point! It is actually meant to add another point. Thus the affordance of this screen is misleading. **[BUG#3]** 

**Desired:**
In the POI list itself, allow user to select a point and simply add it to the route.  (say, long-press on the entry and select the "Add" context menu. Osmand should simply insert that point in the original route.)

**Comment by raindropsfromsky** [comment]
Just occurred to me that a simpler method is to provide a + button against all listed POIs along the route.

Thus each POI will have two buttons instead of one: 
- The + button inserts it in the route as a waypoint.</br>The route is automatically adjusted accommodate the waypoint.
- The X button removes it from the list (this is already there).

**Comment by vshcherb** [comment]
This is not finished UI to add points. Indeed we need to think about adding point quickly to the route to pass by

**Comment by dmpr0** [comment]
Hello! Thank you for sharing your way of using OsmAnd, we try to take into account your feedback.

**Comment by crimean** [comment]
Fixed by https://github.com/osmandapp/Osmand/commit/ac6b6bf788e2205b61fbdd0cb61addcad2fae327
